docs/useStateWithHistory.md

```diff
@@ -0,0 +1,33 @@
+# `useStateHistory`
```

Comment:
```suggestion
# `useStateWithHistory`
```

Reply:
`stateList` is a kinda finite-state machine and works on predefined states list.
While `useStateWithHistory` stores `n` previous states so it is more like fixed size state logger but with travel and modification handles.

I dont think it is possible to combine them, they're about different purposes.